components/brave_vpn/browser/connection/brave_vpn_os_connection_api_unittest.cc

```diff
@@ -0,0 +1,268 @@
+#include "brave/components/brave_vpn/common/pref_names.h"
+#include "build/build_config.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "content/public/test/browser_task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_vpn {
+
+namespace {
+const char kProfileCredentialData[] = R"(
+        {
+          "eap-username": "brave-user",
+          "eap-password": "brave-pwd"
+        }
+      )";
+const char kHostNamesTestData[] = R"([
+        {
+          "hostname": "host-1.brave.com",
+          "display-name": "host-1",
+          "offline": false,
+          "capacity-score": 0
+        },
+        {
+          "hostname": "host-2.brave.com",
+          "display-name": "host-2",
+          "offline": false,
+          "capacity-score": 1
+        },
+        {
+          "hostname": "host-3.brave.com",
+          "display-name": "Singapore",
+          "offline": false,
+          "capacity-score": 0
+        },
+        {
+          "hostname": "host-4.brave.com",
+          "display-name": "host-4",
+          "offline": false,
+          "capacity-score": 0
+        },
+        {
+          "hostname": "host-5.brave.com",
+          "display-name": "host-5",
+          "offline": false,
+          "capacity-score": 1
+        }
+      ])";
+}  // namespace
+
+class BraveVPNOSConnectionAPIUnitTest : public testing::Test {
+ public:
+  BraveVPNOSConnectionAPIUnitTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME) {}
+
+  void SetUp() override {
+    brave_vpn::RegisterLocalStatePrefs(local_pref_service_.registry());
+    connection_api_ = BraveVPNOSConnectionAPI::GetInstanceForTest();
+    connection_api_->SetLocalPrefs(local_state());
+  }
+  PrefService* local_state() { return &local_pref_service_; }
+  BraveVPNOSConnectionAPI* GetConnectionAPI() { return connection_api_.get(); }
+
+ private:
+  TestingPrefServiceSimple local_pref_service_;
+  content::BrowserTaskEnvironment task_environment_;
+  std::unique_ptr<BraveVPNOSConnectionAPI> connection_api_;
+};
+
+// Create os vpn entry with cached connection_info when there is cached
+// connection info.
+TEST_F(BraveVPNOSConnectionAPIUnitTest,
+       CreateOSVPNEntryWithValidInfoWhenConnectTest) {
+  GetConnectionAPI()->CheckConnection();
+
+  // Prepare valid connection info.
+  auto* test_api = static_cast<BraveVPNOSConnectionAPISim*>(GetConnectionAPI());
+  test_api->OnFetchHostnames("region-a", kHostNamesTestData, true);
+  test_api->SetPreventCreationForTesting(true);
+  test_api->OnGetProfileCredentials(kProfileCredentialData, true);
+  EXPECT_TRUE(test_api->connection_info().IsValid());
+  test_api->Connect();
+  base::RunLoop().RunUntilIdle();
+  // With cached connection info, connect process starts with
+  // os vpn entry creation.
+
+  EXPECT_TRUE(test_api->IsConnectionCreated());
+}
+#if !BUILDFLAG(IS_ANDROID)
```

Comment:
IMO, this whole test is only valid for desktop only?

Reply:
we do not have android specific code here, only these  2 tests